src/rx-query.ts

```diff
@@ -490,6 +486,118 @@ export class RxQueryBase<
         this._limitBufferSize = bufferSize;
         return this;
     }
+
+    enablePersistentQueryCache(backend: QueryCacheBackend) {
+        this._queryCacheBackend = backend;
+        this._persistentQueryCacheLoaded = this._restoreQueryCacheFromPersistedState();
+        return this;
+    }
+
+    private async _restoreQueryCacheFromPersistedState() {
+        if (!this._queryCacheBackend) {
+            // no cache backend provided, do nothing
+            return;
+        }
+
+        if (this._persistentQueryCacheResult) {
+            // we already restored the cache once, no need to run twice
+            return;
+        }
+
+        if (this.mangoQuery.skip) {
+            console.error('The persistent query cache only works on non-skip queries.');
+            return;
+        }
+
+        const persistentQueryId = this.persistentQueryId();
+        const value = await this._queryCacheBackend.getItem<string[] | string>(`qc:${persistentQueryId}`);
+
+        const lwt = (await this._queryCacheBackend.getItem(`qc:${persistentQueryId}:lwt`)) as string ?? RX_META_LWT_MINIMUM;
+
+        const primaryPath = this.collection.schema.primaryPath;
+
+        this._persistentQueryCacheResult = value ?? undefined;
+
+        // if this is a regular query, also load documents into cache
+        if (Array.isArray(value) && value.length > 0) {
+            this._persistentQueryCacheResult = value;
+            this._persistentQueryCacheResultLwt = lwt;
+
+            const persistedQueryCacheIds = new Set(this._persistentQueryCacheResult);
+
+            let docsData: RxDocumentData<RxDocType>[] = [];
+
+            // query all docs updated > last persisted, limit to an arbitrary 1_000_000 (10x of what we consider our largest library)
+            const {documents: changedDocs} = await this.collection.storageInstance.getChangedDocumentsSince(
```

Comment:
i am NOT saying we do this now, but it does seem (in theory) fairly wasteful that this LWT will run once for every cached query.

i wonder if you could have something like the docCache that caches all documents `lwt>X`. then if another query is trying to fetch `lwt>Y` (Y>X) you can just loop through lwt>X. if another query tries to fetch `lwt>Z` (X>Z), you filter the db for `X>lwt>Z` only, and get the rest from `lwt>X` :P

Reply:
Hm, I think you still need to run `lwt > last_of(set_X)` to also fetch updates since the first query ran. But in general I agree, we could do this per collection instead of per query.

Reply:
it's also a bit strange that the memory query cache is a property of the RX collection, but the persistent query cache is a property of each individual query. might could kill two birds with one stone if you get around to it (but not blocking this PR)

Reply:
I think we can do it, but I believe it is not super simple. For example, we need to block (enqueue) and dedup the retrieval across all queries that query the same collection. When a query with an older timestamp starts after a query with a newer timestamp, we need to cancel the request and initiate a new one.